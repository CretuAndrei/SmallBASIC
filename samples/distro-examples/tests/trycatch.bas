? "start"
try
  ? "inner try"
  if (1==1) then
    try 
    catch 
      ? "should never be printed"
    end try
    if (2==2) then
      try
       open "com2000:" AS #1
       try 
         ? "should never be printed"
       catch 
         ? "should never be printed"
       end try
       ? "should never be printed"
      catch "err"
       ? "should never be printed"
      catch err
       ? "open failed", err
      end try
      ? "after try
      throw "an error has occurred"
      ? "should never be printed"
    fi
  fi
  try 
   ? "should never be printed"
  catch 
   ? "should never be printed"
  end try
  open "com2000:" AS #1
  ? "should never be printed"
catch "some error that is not thrown"
  ? "should never be printed"
catch "an error has occurred"
  ? "catch by error name"
end try

? "outer after try"


