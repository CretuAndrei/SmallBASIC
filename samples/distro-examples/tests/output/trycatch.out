start
inner try
open failed	FS(2): NO SUCH FILE OR DIRECTORY
after try
catch by error name
outer after try
Stack:
 TRY: 199
 TRY: 198
 TRY: 197
 TRY: 196
 TRY: 195
 IF: 194
 IF: 193
 IF: 192
 SUB: 216


 * RTE-ERROR AT ../../../samples/distro-examples/tests/trycatch.bas:200 * 
Description:
Division by zero

